src/main/kotlin/device/audio/AudioPlayer.kt

```diff
@@ -0,0 +1,21 @@
+import javafx.scene.media.AudioClip
+import java.net.URI
+
+
+class AudioPlayer(audioClip: AudioClip) {
```

Comment:
AudioPlayer is meant to be a replacement for AudioClip in our app (because AudioClip is an implementation detail specific to the device you're on; Android uses AudioTrack). After all, if the developer already has an AudioClip object to use for constructing this, why wouldn't they just use AudioClip directly?

The API you're defining is to take in a:
- File
- URI (this is really more optional, but I'm fine with it being an option)
- String containing a path to a file

These are going to be defined in your interface, and implemented here.
You also want to use the constructor to take the opportunity for initializing your AudioClip. Since we have it there, there's no reason to make it a nullable. 

Nullable means the possibility of null pointer exceptions, the most common bug in all of software development. Kotlin forces you to do a null check to prevent this, but your elvis operators will mean that the code like play becomes a no-op if the player is null. Not playing when you call play though is a bug, but now you won't throw an exception, so the code doesn't believe something went wrong. We won't get a log of the stacktrace, and our user doesn't have working software.

We should be allergic to nullable types. There's a huge cost to allowing them, and it takes huge justification. If our current design/architecture won't allow us to avoid nullable types, we should first consider if there is another way of architecting the system so that they are no longer required. Only when we have exhausted all options should nullable be allowed.